app/models/spree/order.rb

```diff
@@ -639,8 +639,6 @@ def fee_handler
     end
 
     def process_each_payment
-      raise Core::GatewayError, Spree.t(:no_pending_payments) if pending_payments.empty?
```

Comment:
Hmmm... I think this could have unintended consequences.

Reply:
That's a good point, I'll look into what happens in that case. We might need to add a test or two around it if it's doing something unexpected since it's not currently being caught by the test suite.

Reply:
Ultimately, IMO, we need to remove the responsibility of processing the payments from the order model. What this issue/PR are about is decoupling these two concerns thus, not making every piece of business logic depend on this model.

Reply:
It looks there was already a mechanism to skip this processing.

https://github.com/openfoodfoundation/openfoodnetwork/blob/0d30e3359f0238ca0e4b2c670cf4e3b2da717be5/app/models/spree/order/checkout.rb#L70-L74

I'm not judging if it's a good idea but have we investigated this? was this introduced for subs?